SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialPackageMapClient.cpp

```diff
@@ -418,21 +418,28 @@ void FSpatialNetGUIDCache::RemoveEntityNetGUID(Worker_EntityId EntityId)
 		return;
 	}
 
-	const FClassInfo& Info = SpatialNetDriver->ClassInfoManager->GetOrCreateClassInfoByClass(UnrealMetadata->GetNativeEntityClass());
-
 	SpatialGDK::TSchemaOption<FUnrealObjectRef>& StablyNamedRefOption = UnrealMetadata->StablyNamedRef;
 
-	for (auto& SubobjectInfoPair : Info.SubobjectInfo)
+	if (UnrealMetadata->NativeClass.IsStale())
 	{
-		FUnrealObjectRef SubobjectRef(EntityId, SubobjectInfoPair.Key);
-		if (FNetworkGUID* SubobjectNetGUID = UnrealObjectRefToNetGUID.Find(SubobjectRef))
-		{
-			NetGUIDToUnrealObjectRef.Remove(*SubobjectNetGUID);
-			UnrealObjectRefToNetGUID.Remove(SubobjectRef);
+		UE_LOG(LogSpatialPackageMap, Warning, TEXT("Attempting to remove stale object from package map - %s"), *UnrealMetadata->ClassPath);
```

Comment:
Anything the user can do to prevent this warning once they see it?

Reply:
No, I believe this only happens when clients unload a level. This `fix` is really just a stop gap until we re-work this function (UNR-1807)

Reply:
In that case maybe we should downgrade to a log?

Reply:
My concern is that there are other reasons that this could happen, so would prefer this to be explicit until we resolve it properly.

Reply:
That makes sense.